packages/babel-core/test/config-loading.js

```diff
@@ -188,25 +188,26 @@ describe("@babel/core config loading", () => {
       const options2 = loadConfig(opts).options;
       expect(options2.plugins.length).toBe(options1.plugins.length);
 
+      expect(options2.plugins[0]).not.toBe(options1.plugins[0]);
       for (let i = 0; i < options1.plugins.length; i++) {
         if (i === 0) {
-          expect(options2.plugins[i]).not.toBe(options1.plugins[i]);
-        } else {
-          expect(options2.plugins[i]).toBe(options1.plugins[i]);
+          continue;
         }
+        expect(options2.plugins[i]).toBe(options1.plugins[i]);
       }
 
       process.env.INVALIDATE_PLUGIN3 = true;
 
       const options3 = loadConfig(opts).options;
       expect(options3.plugins.length).toBe(options1.plugins.length);
-
+      for (const i of [0, 5]) {
+        expect(options3.plugins[i]).not.toBe(options1.plugins[i]);
+      }
       for (let i = 0; i < options1.plugins.length; i++) {
         if (i === 0 || i === 5) {
-          expect(options3.plugins[i]).not.toBe(options1.plugins[i]);
-        } else {
-          expect(options3.plugins[i]).toBe(options1.plugins[i]);
+          continue;
```

Comment:
Here we can first assert that `options1.plugins.length` is `6`, and then either iterate from `i = 1` to `i <= 5`, or manually write
```js
      expect(options3.plugins[1]).not.toBe(options1.plugins[1]);
      expect(options3.plugins[2]).not.toBe(options1.plugins[2]);
      expect(options3.plugins[3]).not.toBe(options1.plugins[3]);
      expect(options3.plugins[4]).not.toBe(options1.plugins[4]);
```

Similar suggestions applies also to the next parts of this file.